sympy/functions/elementary/trigonometric.py

```diff
@@ -1568,6 +1568,101 @@ def taylor_term(n, x, *previous_terms):
                     bernoulli(2*k)*x**(2*k - 1)/factorial(2*k))
 
 
+class sinc(TrigonometricFunction):
+    r"""Represents unnormalized sinc function
+
+    Examples
+    ========
+
+    >>> from sympy import sinc, oo, jn, Product, Symbol
+    >>> from sympy.abc import x
+    >>> sinc(x)
+    sinc(x)
+
+    * Automated Evaluation
+
+    >>> sinc(0)
+    1
+    >>> sinc(oo)
+    0
+
+    * Differentiation
+
+    >>> sinc(x).diff()
+    (x*cos(x) - sin(x))/x**2
+
+    * Series Expansion
+
+    >>> sinc(x).series()
+    1 - x**2/6 + x**4/120 + O(x**6)
+
+    * As zero'th order spherical Bessel Function
+
+    >>> sinc(x).rewrite(jn)
+    jn(0, x)
+
+    * Product form
+
+    >>> k = Symbol('k', integer=True, positive=True)
+    >>> sinc(x).rewrite(Product)
+    Product(2**(-k)*cos(x), (k, 1, oo))
+
+    References
+    ==========
+
+    .. [1] http://en.wikipedia.org/wiki/Sinc_function
+    .. [2] Euler, Leonhard (1735). "On the sums of series of reciprocals"
+
+    """
+
+    def fdiff(self, argindex=1):
+        x = self.args[0]
+        if argindex == 1:
+            return (x*cos(x) - sin(x)) / x**2
+        else:
+            raise ArgumentIndexError(self, argindex)
+
+    @classmethod
+    def eval(cls, arg):
+        if arg.is_zero:
+            return S.One
+        if arg.is_Number:
+            if arg in [S.Infinity, -S.Infinity]:
+                return S.Zero
+            elif arg is S.NaN:
+                return S.NaN
+
+        if arg is S.ComplexInfinity:
+            return S.NaN
+
+        if arg.could_extract_minus_sign():
+            return cls(-arg)
+
+        pi_coeff = _pi_coeff(arg)
+        if pi_coeff is not None:
+            if pi_coeff.is_integer:
+                if arg.is_nonzero:
+                    return S.Zero
+            elif (2*pi_coeff).is_integer:
+                return S.NegativeOne**(pi_coeff - S.Half) / arg
+
+    def _eval_nseries(self, x, n, logx):
+        x = self.args[0]
+        return (sin(x)/x)._eval_nseries(x, n, logx)
+
+    def _eval_rewrite_as_jn(self, arg):
+        from sympy.functions.special.bessel import jn
+        return jn(0, arg)
+
+    def _eval_rewrite_as_sin(self, arg):
+        return sin(arg) / arg
+
+    def _eval_rewrite_as_Product(self, arg):
+        from sympy import Product
+        k = Symbol('k', Dummy=True, integer=True, positive=True)
+        return Product(cos(arg)/2**k, (k, 1, S.Infinity))
```

Comment:
Perhaps the formula should not belong to this PR at all. There are product formulas for many other functions, but they have not been included in SymPy. They probably do not fit well into its design.